app/src/sharedTest/java/org/oppia/android/app/profile/AdminAuthActivityTest.kt

```diff
@@ -501,26 +540,77 @@ class AdminAuthActivityTest {
         AdminAuthEnum.PROFILE_ADMIN_CONTROLS.value
       )
     ).use {
-      onView(allOf(withId(R.id.input), isDescendantOfA(withId(R.id.admin_auth_input_pin)))).perform(
-        typeText("12354"),
-        pressImeActionButton()
-      )
       onView(
         allOf(
-          withId(R.id.error_text),
+          withId(R.id.admin_auth_input_pin_edit_text),
           isDescendantOfA(withId(R.id.admin_auth_input_pin))
         )
-      ).check(matches(withText(context.resources.getString(R.string.admin_auth_incorrect))))
+      ).perform(
+        appendText("12354"),
+        pressImeActionButton()
+      )
+      onView(withId(R.id.admin_auth_input_pin))
+        .check(matches(hasErrorText(R.string.admin_auth_incorrect)))
```

Comment:
hasErrorText does'nt work for InputTextLayout 
suggested solutions

- This work but I'm not sure if it is a good practice
```
      onView(withId(R.id.admin_auth_input_pin))
        .check(matches(hasDescendant(withText(R.string.admin_auth_incorrect))))
```

 - add a custom matcher specifically for InputTextLayout 
    ref  - https://stackoverflow.com/a/56598890/12088301

Reply:
I'm sorry i was a bit confused the testcase works well 
may be ime_actions needs to be updated as mentioned by rajat 
please mark this conversation as resolved